### PR TITLE
Adds run interpolation script

### PR DESCRIPTION
**Pull request by rodrigonogueira4**
Interpolate two runs by first normalizing their query-level scores (between 0 and 1) and then doing:
alpha * run1_score + (1 - alpha) * run2_score.

### CODE REVIEW

File: src/main/python/interpolate.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+'''
+Anserini: A Lucene toolkit for replicable information retrieval research
```
Comment (lintool) [inline src/main/python/interpolate.py:3]: mangled boilerplate... use https://github.com/castorini/anserini/blob/master/src/main/python/run_es_regression.py#L1

File: src/main/python/interpolate.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+def load_run(path):
+    run = collections.defaultdict(dict)
+    for line in open(path):
+        query_id, _, doc_id, _, score, _ = line.split(' ')
+        run[query_id][doc_id] = float(score)
+
+    return run
+
+
+parser = argparse.ArgumentParser(
```
Comment (lintool) [inline src/main/python/interpolate.py:37]: If we don't have qrels, add an ability to just output a set alpha?

File: src/main/python/interpolate.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    description='Interpolate 2 runs by first normalizing their scores (between'
+                '0 and 1) and then doing: alpha * run1_score + (1 - alpha) * '
+                'run2_score.')
```
Comment (lintool) [inline src/main/python/interpolate.py:40]: Are you setting width=100? These lines seem really short to need linebreaks...

File: src/main/python/interpolate.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    print('alpha:', alpha, 'score:', score)
+
+    if score > best_score:
+        best_run = run_fusion
+        best_score = score
+        best_alpha = alpha
+
+print('best_alpha:', best_alpha, 'best_score:', best_score)
+
+with open(args.output, 'w') as fout:
```
Comment (lintool) [inline src/main/python/interpolate.py:92]: It appears that you don't require the doc to be in _both_ runs, right? If so, the union may yield more hits?

If so, maybe add a `-k` parameter to trim down to a set `k`, e.g., 1000?

File: src/main/python/interpolate.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    for query_id, doc_id_scores in best_run.items():
+        for rank, (doc_id, score) in enumerate(doc_id_scores.items()):
+            rank += 1
+            fout.write(f'{query_id} Q0 {doc_id} {rank} {score} Fusion\n')
```
Comment (lintool) [inline src/main/python/interpolate.py:96]: Can we make `Fusion` into a `-runtag` param? We need this to generate final TREC runs...

File: src/main/python/interpolate.py (hunk 3 above)
Comment (rodrigonogueira4) [reply src/main/python/interpolate.py:40]: I'm using PEP8, which is 80 chars

Comment (lintool) [reply src/main/python/interpolate.py:40]: > Some teams strongly prefer a longer line length. For code maintained exclusively or primarily by a team that can reach agreement on this issue, it is okay to increase the nominal line length from 80 to 100 characters (effectively increasing the maximum length to 99 characters), provided that comments and docstrings are still wrapped at 72 characters.

https://pep8.org/#maximum-line-length

haha, can we agree? ;)

File: src/main/python/interpolate.py (hunk 5 above)
Comment (rodrigonogueira4) [reply src/main/python/interpolate.py:96]: Sure

File: src/main/python/interpolate.py (hunk 1 above)
Comment (rodrigonogueira4) [reply src/main/python/interpolate.py:3]: Interesting... if we just copy and paste, the line breaks disappear. Anyway, fixed!

File: src/main/python/interpolate.py (hunk 3 above)
Comment (rodrigonogueira4) [reply src/main/python/interpolate.py:40]: haha, ok, I do prefer longer lines

File: src/main/python/interpolate.py (hunk 2 above)
Comment (rodrigonogueira4) [reply src/main/python/interpolate.py:37]: Sure

File: src/main/python/interpolate.py (hunk 4 above)
Comment (rodrigonogueira4) [reply src/main/python/interpolate.py:92]: Sure

**Comment by rodrigonogueira4** [comment]
@lintool, PTAL

**Pull request merged**